File: OOP/7_Function_Overloading.py

# Overloading allows multiple functions with the same name to coexist in the same scope, differentiated by their parameter types or counts.

class Calculator:
    def sum(self,num1,num2,num3=None):
       if num3 is None:
           return num1+num2
       return num1+num2+num3
    
    
cal = Calculator()
print(cal.sum(1,2,3))
print(cal.sum(1,2))

File: OOP/7_Function_Overloading.cpp
#include <bits/stdc++.h>
using namespace std;

class Caluculator
{
public:
// function aita jeta 2 ta argument nei
    int sum(int num1, int num2)
    {
        return num1 + num2;
    }

    // same name er function overloaded hoye 3 ta argumetn nei 

    int sum(int num1, int num2, int num3)
    {
        return num1 + num2 + num3;
    }
};
int main()
{
    Caluculator obj;

    cout << obj.sum(1, 2) << endl;
    cout << obj.sum(1, 2, 3) << endl;

    return 0;
}
